packages/ra-core/src/sideEffect/accumulate.js

```diff
@@ -4,57 +4,61 @@ import { call, cancel, fork, put, takeEvery } from 'redux-saga/effects';
 /**
  * Example
  *
- * let debouncedIds = {
+ * let accumulations = {
  *   posts: { 4: true, 7: true, 345: true },
  *   authors: { 23: true, 47: true, 78: true },
  * }
  */
-const debouncedIds = {};
-const addIds = (resource, ids) => {
-    if (!debouncedIds[resource]) {
-        debouncedIds[resource] = {};
-    }
-    ids.forEach(id => {
-        debouncedIds[resource][id] = id;
-    }); // fast UNIQUE
+let accumulations = {};
+
+const addIds = (accumulations, { payload: { ids } }) => {
+    const accumulatedValue = new Set(accumulations) || new Set();
+    ids.forEach(id => accumulatedValue.add(id));
+    return Array.from(accumulatedValue);
 };
-const getIds = resource => {
-    const ids = Object.values(debouncedIds[resource]);
-    delete debouncedIds[resource];
-    return ids;
+
+const getAccumuledValue = key => {
+    const accumulatedValue = accumulations[key];
+    delete accumulations[key];
+    return accumulatedValue;
 };
 
 const tasks = {};
 
 /**
- * Fetch the list of accumulated ids after a delay
+ * Fetch the accumulated value after a delay
  *
  * As this gets canceled by subsequent calls to accumulate(), only the last
  * call to finalize() will not be canceled. The delay acts as a
  * debounce.
  *
  * @see https://redux-saga.js.org/docs/recipes/#debouncing
  */
-function* finalize(resource, actionCreator) {
+export function* finalize(key, actionCreator) {
     // combined with cancel(), this debounces the calls
     yield call(delay, 50);
-    yield put(actionCreator(resource, getIds(resource)));
-    delete tasks[resource];
+    const accumulatedValue = getAccumuledValue(key);
+    const action = actionCreator(key, accumulatedValue);
```

Comment:
why create a `const` here? It makes your diff harder to read for no good reason.